mpas_analysis/shared/climatology/climatology.py

```diff
@@ -608,17 +806,23 @@ def ds_to_weights(ds):
             for var in ds.data_vars:
                 weights[var] = ds[var].notnull()
         else:
-            raise TypeError('ds must be an instance of either xarray.Dataset or xarray.DataArray.')
+            raise TypeError('ds must be an instance of either xarray.Dataset '
+                            'or xarray.DataArray.')
 
         return weights
 
-    dsWeightedSum = (ds * ds.daysInMonth).sum(dim='Time', keep_attrs=True)
+    if maskVaries:
+        dsWeightedSum = (ds * ds.daysInMonth).sum(dim='Time', keep_attrs=True)
 
-    weights = ds_to_weights(ds)
+        weights = ds_to_weights(ds)
 
-    weightSum = (weights * ds.daysInMonth).sum(dim='Time')
+        weightSum = (weights * ds.daysInMonth).sum(dim='Time')
 
-    timeMean = dsWeightedSum / weightSum.where(weightSum > 0.)
+        timeMean = dsWeightedSum / weightSum.where(weightSum > 0.)
+    else:
+        days = ds.daysInMonth.sum(dim='Time')
+        dsWeightedSum = (ds * ds.daysInMonth).sum(dim='Time', keep_attrs=True)
+        timeMean = dsWeightedSum / days.where(days > 0.)
```

Comment:
@pwolfram, the first sum in the else case (over daysInMonth) is over a very small vector of dimension `Time`, so should typically take negligible time.

I think making 5% errors that are totally avoidable is completely inexcusable, even in the interest of improved performance.  In this case, I expect performance to be negligibly different from calling `mean` except perhaps for some lack of optimization that i feel sure could be addressed.